src/cpl/share_esmf/PrigentRoughnessStreamType.F90

```diff
@@ -0,0 +1,316 @@
+              stream_yearLast     = 1997,                                               &
+              stream_yearAlign    = 1,                                                  &
+              stream_offset       = 0,                                                  &
+              stream_taxmode      = 'extend',                                           &
+              stream_dtlimit      = 1.0e30_r8,                                          &
+              stream_tintalgo     = 'linear',                                           &
+              stream_name         = 'Prigent roughness',                                 &
+              rc                  = rc)
+         if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) then
+            call ESMF_Finalize(endflag=ESMF_END_ABORT)
+         end if
+
+         ! Explicitly set current date to a hardcoded constant value. Otherwise
+         ! using the real date can cause roundoff differences that are
+         ! detrected as issues with exact restart.  EBK M05/20/2017
+         ! call get_curr_date(year, mon, day, sec)
+         year = 1997
+         mon  = 12
+         day  = 31
+         sec  = 0
+         mcdate = year*10000 + mon*100 + day
+
+         call shr_strdata_advance(sdat_rghn, ymd=mcdate, tod=sec, logunit=iulog, istr='prigentrghn', rc=rc)
+         if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) then
+            call ESMF_Finalize(endflag=ESMF_END_ABORT)
+         end if
+
+         ! Get pointer for stream data that is time and spatially interpolate to model time and grid
+         do n = 1,size(stream_varnames) 
+            call dshr_fldbun_getFldPtr(sdat_rghn%pstrm(1)%fldbun_model, stream_varnames(n), fldptr1=dataptr1d, rc=rc)
+            if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) then
+               call ESMF_Finalize(endflag=ESMF_END_ABORT)
+            end if
+            if (trim(stream_varnames(n)) == 'Z0a') then
+               ig = 0
+               do g = bounds%begg,bounds%endg
+                  ig = ig+1
+                  this%prigent_rghn(g) = dataptr1d(ig)
+               end do
+
+            end if
+
+         end do
+      end if
+   !end if         !commented out
+
+  end subroutine Init
+
+  !==============================================================================
+  logical function UseStreams(this)
+    !
+    ! !DESCRIPTION:
+    ! Return true if
+    !
+    ! !USES:
+    !
+    ! !ARGUMENTS:
+    implicit none
+    class(prigentroughnessstream_type) :: this
+    !
+    ! !LOCAL VARIABLES:
+    if ( trim(control%stream_fldFileName_prigentroughness) == '' )then
+       UseStreams = .false.  ! this won't happen and UseStreams will always be true
+    else
+       UseStreams = .true.
+    end if
+  end function UseStreams
```

Comment:
I'm going to add a check routine to make sure the ReadNML was called before you call UseStreams.

Reply:
done